interfaces/builtin/all_test.go

```diff
@@ -332,6 +332,20 @@ apps:
         plugs: [iface]
 `
 
+const testInvalidSlotInterfaceYaml = `
+name: testsnap
+slots:
+ foo:
+  interface: iface
+`
```

Comment:
Can you please include an app and a hook here as well. Make them use the slot.

Nitpick: please rename `foo` to just `iface`.